egs/wsj/s5/steps/nnet3/multi/make_tdnn_configs.py

```diff
@@ -0,0 +1,399 @@
+          sep_inputs=[]
+          tot_splices = [ ('Offset(renorm{0}, {1})'.format(l-1, n) if n !=0 else 'renorm{0}'.format(l-1))
+                      for n in splice_array[l-1] ]
+
+          for i in range(0, num_outputs):
+            splices = [ ('Offset(renorm{0}-{2}, {1})'.format(l-1, n, i) if n !=0 else 'renorm{0}-{1}'.format(l-1, i))
+                        for n in splice_array[l-1] ]
+            sep_inputs.append('IfDefined(Append({0}))'.format(', '.join(splices)))
+          ivec_input = 'Sum({0})'.format(', '.join(sep_inputs))
+          cur_input = 'Append({0}, {1})'.format(','.join(tot_splices), ivec_input)
+        else:
+          splices = [ ('Offset(renorm{0}, {1})'.format(l-1, n) if n !=0 else 'renorm{0}'.format(l-1))
+                      for n in splice_array[l-1] ]
+          cur_input='Append({0})'.format(', '.join(splices))
+    if l == 1 and separate_init_layer_per_output and num_outputs > 1:
+      splices = [ ('Offset(input, {0})'.format(n) if n != 0 else 'input') for n in splice_array[l-1] ]
+      cur_input='Append({0})'.format(', '.join(splices))
+      print('component-node name=affine{0} component=affine{0} input={1} '.
+            format(l, cur_input), file=f)
+      print('component-node name=nonlin{0} component=nonlin{0} input=affine{0}'.
+            format(l), file=f)
+      print('component-node name=renorm{0} component=renorm{0} input=nonlin{0}'.
+            format(l), file=f)
+      orig_input='Append({0})'.format(', '.join(splices))
+      for i in range(0, num_outputs):
+        if args.ivector_dim > 0: ivec_input= ('ReplaceIndex(ivector-{0}, t, 0)'.format(i))
+         
+        print('component-node name=affine{0}-{2} component=affine{0}-{2} input={1}'.
+              format(l, ivec_input, i), file=f)
+        print('component-node name=nonlin{0}-{1} component=nonlin{0}-{1} input=affine{0}-{1}'.
+              format(l, i), file=f)
+        print('component-node name=renorm{0}-{1} component=renorm{0}-{1} input=nonlin{0}-{1}'.
+              format(l,i), file=f)
+        print('component-node name=final-affine-{1} component=final-affine-{1} input=Append(IfDefined(renorm{0}-{1}), renorm{0})'.
+              format(l, i), file=f)
+    else:
+      print('component-node name=affine{0} component=affine{0} input={1} '.
+            format(l, cur_input), file=f)
+      print('component-node name=nonlin{0} component=nonlin{0} input=affine{0}'.
+            format(l), file=f)
+      print('component-node name=renorm{0} component=renorm{0} input=nonlin{0}'.
+            format(l), file=f)
+      if (num_outputs > 1):
+        for i in range(0, num_outputs):
+          print('component-node name=final-affine-{1} component=final-affine-{1} input=renorm{0}'.
+                format(l, i), file=f)
+      else:
+        print('component-node name=final-affine component=final-affine input=renorm{0}'.
+              format(l), file=f)
+
+    if args.include_log_softmax == "true":
```

Comment:
Ideally, you should use the format used in steps/nnet3/tdnn/make_configs.py.
That is the latest version of the script that is being used. You can use functions like 
nodes.AddFinalLayer, nodes.AddAffRelNormLayer which would be more readable.